src/inmanta/ast/constraint/expression.py

```diff
@@ -77,7 +77,6 @@ def __init__(self, attr: Optional[Reference], name: LocatableString) -> None:
     def requires_emit(self, resolver: Resolver, queue: QueueScheduler) -> Dict[object, ResultVariable]:
         # introduce temp variable to contain the eventual result of this stmt
         temp = ResultVariable()
-        temp.set_provider(self)
         # construct waiter
         resumer = IsDefinedReferenceHelper(temp, self.attr, self.name)
```

Comment:
shouldn't the `isDefinedReferenceHelper` also put a promise on temp, to replace the `set_provider`